types/src/transaction/error.rs

```diff
@@ -242,7 +248,10 @@ impl Decodable for Error {
             ERROR_ID_INVALID_ASSET_TYPE => Error::InvalidAssetType(rlp.val_at(1)?),
             ERROR_ID_SCRIPT_HASH_MISMATCH => Error::ScriptHashMismatch(rlp.val_at(1)?),
             ERROR_ID_INVALID_SCRIPT => Error::InvalidScript,
-            ERROR_ID_FAILED_TO_UNLOCK => Error::FailedToUnlock(rlp.val_at(1)?),
+            ERROR_ID_FAILED_TO_UNLOCK => Error::FailedToUnlock {
```

Comment:
It might be good to check `rlp.item_count()` here. (I think every other error code should check it also)